src/HotChocolate/Fusion-vnext/test/Fusion.Execution.Tests/Planning/InlineFragmentOperationRewriterTests.cs

```diff
@@ -237,4 +236,45 @@ fragment Product on Product {
             }
             """);
     }
+
+    [Test]
+    public void Leafs_With_Different_Directives_Do_Not_Merge()
+    {
+        // arrange
+        var compositeSchemaDoc = Utf8GraphQLParser.Parse(FileResource.Open("fusion1.graphql"));
+        var compositeSchema = CompositeSchemaBuilder.Create(compositeSchemaDoc);
+
+        var doc = Utf8GraphQLParser.Parse(
+            """
+            query($skip: Boolean!) {
+                productById(id: 1) {
+                    ... Product
+                }
+            }
+
+            fragment Product on Product {
+                id
+                name @include(if: $skip)
```

Comment:
@tobias-tengler I have reworked the field merging with what we discussed. It needs more tests and especially for edge cases.

Reply:
We actually could handle static includes and skips at this level already. This way we would not need to deal with them in the planning process.

Reply:
Awesome, thank you! I haven't had the time to work on it myself yet. I've started on tests though, so once this is in I can take over and work on the edge cases / tests.